tests/test_config.py

```diff
@@ -0,0 +1,97 @@
+import shlex
+import pytest
+import mock
+
+
+from jrnl.args import parse_args
+from jrnl.jrnl import run
+from jrnl import install
+from jrnl import jrnl
+
+
+@pytest.fixture()
+def minimal_config():
+    cfg = {
+        "colors": {"body": "red", "date": "green"},
+        "default": "/tmp/journal.jrnl",
+        "editor": "vim",
+        "encrypt": False,
+        "journals": {"default": "/tmp/journals/journal.jrnl"},
+    }
+    yield cfg
+
+
+@pytest.fixture()
+def expected_override(minimal_config):
+    exp_out_cfg = minimal_config.copy()
+    exp_out_cfg["editor"] = "nano"
+    exp_out_cfg["journal"] = "/tmp/journals/journal.jrnl"
+    yield exp_out_cfg
+
+
+@mock.patch("sys.stdin.isatty")
+@mock.patch.object(install, "load_or_install_jrnl")
+@mock.patch("subprocess.call")
+def test_override_configured_editor(
+    mock_subprocess_call,
+    mock_load_or_install,
+    mock_isatty,
+    minimal_config,
+    expected_override,
+    capsys,
+):
+    mock_load_or_install.return_value = minimal_config
+    mock_isatty.return_value = True
+
+    cli_args = shlex.split('--config-override editor "nano"')
```

Comment:
I think we should we be using os_compat split_args instead of shlex.split to be cross-platform

Reply:
good to know! For the purpose of this test, however, this is moot since I ended up deleting test_config.py altogether